packages/opentelemetry-metrics/src/export/types.ts

```diff
@@ -0,0 +1,370 @@
+   * then the start_time should also be reset.
+   */
+  COUNTER_INT64,
+  /**
+   * Floating point counter measurement. The value cannot decrease, if
+   * resets then the start_time should also be reset. Recorded values are
+   * always >= 0.
+   */
+  COUNTER_DOUBLE,
+  /**
+   * Histogram cumulative measurement. The count and sum cannot decrease,
+   * if reset then the start_time should also be reset.
+   */
+  CUMULATIVE_HISTOGRAM,
+  /**
+   * Some frameworks implemented Histograms as a summary of observations
+   * (usually things like request durations and response sizes). While it
+   * also provides a total count of observations and a sum of all observed
+   * values, it calculates configurable percentiles over a sliding time
+   * window. This is not recommended, since it cannot be aggregated.
+   */
+  SUMMARY,
+}
+
+/**
+ * A collection of data points that describes the time-varying values
+ * of a metric.
+ */
+export interface TimeSeries {
+  /**
+   * The set of label values that uniquely identify this timeseries. Applies to
+   * all points. The order of label values must match that of label keys in the
+   * metric descriptor.
+   */
+  readonly labelValues: LabelValue[];
+
+  /**
+   * The data points of this timeseries. Point.value type MUST match the
+   * MetricDescriptor.type.
+   */
+  readonly points: Point[];
+}
+
+/** The LabelValue type. null value indicates an unset. */
+export interface LabelValue {
+  /** The value for the label. */
+  readonly value: string | null;
+}
+
+/** A timestamped measurement. */
+export interface Point {
+  /**
+   * Must be present for counter/cumulative metrics. The time when the
+   * cumulative value was reset to zero. The cumulative value is over the time
+   * interval (start_timestamp, timestamp]. If not specified, the backend can
+   * use the previous recorded value.
+   */
+  readonly startTimestamp?: Timestamp;
+
+  /**
+   * The moment when this point was recorded. Inclusive.
+   * If not specified, the timestamp will be decided by the backend.
+   */
+  readonly timestamp: Timestamp;
+
+  /**
+   * The actual point value.
+   * 64-bit integer or 64-bit double-precision floating-point number
+   * or distribution value
+   * or summary value. This is not recommended, since it cannot be aggregated.
+   */
+  readonly value: number | HistogramValue | SummaryValue;
+}
+
+/**
+ * Histograms contains summary statistics for a population of values. It
+ * optionally contains a histogram representing the distribution of those
+ * values across a set of buckets.
+ */
+export interface HistogramValue {
+  /**
+   * The number of values in the population. Must be non-negative. This value
+   * must equal the sum of the values in bucket_counts if a histogram is
+   * provided.
+   */
+  readonly count: number;
+  /**
+   * The sum of the values in the population. If count is zero then this field
+   * must be zero.
+   */
+  readonly sum: number;
```

Comment:
Does the spec say this should always be required? For something like a request latency histogram, I wouldn't think the sum would be very usefl.

Reply:
Right now, there is only the proto definition but when I look at https://github.com/census-instrumentation/opencensus-node/blob/master/packages/opencensus-core/src/metrics/export/types.ts#L274 you are right. It's just not marked as optional. Will fix.